### DISCUSSION TITLE
Electron-like-with-Servo

### DISCUSSION THREAD

**Discussion by NetOpWibby** [servo/servo#28411] [Ideas]
"Starling" was apparently something being worked on in 2018 after the ending of BrowserHTML but I see no mention of it online, unless my search-fu is terrible.

I'd much prefer to use Servo than Electron but it's not as easy to get up and running as Electron. Are there public examples of a Servo-based app built with HTML/CSS/JS?

**Comment by jdm** [comment]
There are no such apps at this time. The only examples of embedding Servo right now are https://github.com/servo/servo/tree/master/ports and https://github.com/servo/servo/tree/master/support.

**Comment by pwnorbitals** [comment]
(Triage : issue can probably be closed now)

**Comment by NetOpWibby** [comment]
Sadness

**Comment by sandstrom** [comment]
I think embedding (something similar to WebViews or Electron) may be an interesting beachhead for Servo. In other words, making it possible to swap in Servo into Electron as the engine, or to use Servo on its own in a fashion similar to Electron.

Since embedding often (though not always) will have a single set of CSS and JS that it's expected to run (basically a bundled app), there is less problem caused by e.g. servo not supporting all Web APIs, or not handling all edge cases. Basically, if it runs *your* code, that's enough.

So it'll probably be "useful" (regardless of where you put the threshold) earlier in an embedding context, compared to it being "useful" as a general browser.

Should it catch on as a tool for embedding, that would drive more usage and more contributions, which would be a beneficial circle.

**Comment by WhyNotHugo** [comment]
I think this is a big fault of servo, and one of the big reasons why chromium remains so dominant.

I've a web app that I'd like to bundle into a desktop program, and I'd quite like to use servo for this. Is there any documentation or alike on this, or is it totally unsupported?

**Reply by NetOpWibby** [reply]
Mozilla has zero interest in maintaining Servo and let go a LOT of people working on it. The best thing you could do is use Tauri instead of Electron.

**Comment by renhiyama** [comment]
Mozilla go brrr....
Never expected this from a non profit organization whose work was to promote open source... They just abandoned it after they got what they needed?

**Comment by CrezyDud** [comment]
no

**Comment by makoConstruct** [comment]
Tauri-servo now has nlnet funding https://nlnet.nl/project/Tauri-Servo/

> This project would incorporate Servo into the Tauri project, enabling it to run applications in a consistent, open source web runtime on major desktop and mobile platforms. In doing so, the project would also identify and address the highest priority web compatibility issues in Servo, while preparing a roadmap for significant compatibility issues that remain unaddressed.

**Reply by NetOpWibby** [reply]
Would be nice if there was an official site

**Reply by Visne** [reply]
Where is this development actually happening?

**Reply by chee** [reply]
https://github.com/servo/servo/issues/30593

**Comment by NetOpWibby** [comment]
BUMP

**Reply by jdm** [reply]
Comments like this are not constructive. If you have a question, ask it.

**Comment by NetOpWibby** [comment]
What's the progress on this, if any?

**Reply by jdm** [reply]
https://servo.org/blog/2024/01/19/embedding-update/

**Reply by jdm** [reply]
https://github.com/tauri-apps/wry/tree/servo-wry-demo

**Reply by NetOpWibby** [reply]
Damn, the last update was last year. Oh well.
